### PR TITLE
Add Related module to test

### PR DESCRIPTION
**Pull request by cwg83**

### CODE REVIEW

File: client/src/components/Related/ProductCard.jsx [hunk 1]
```diff
@@ -1,42 +1,66 @@
...
 const RelatedProduct = (props) => {
+  const navigate = useNavigate();
   const [averageRating, setAverageRating] = useState(0);
+  const stars = props.generateStars(averageRating, 'related');
 
+  // const isInYourOutfit = () => {
+  //   return props.yourOutfit.filter((product) => product.id === props.product.id).length > 0
+  // }
+
+  const routeChange = () => {
```
Comment (cwg83) [inline client/src/components/Related/ProductCard.jsx:14]: Function to change the URL to the new product ID when a related product is clicked. Should update state in App.jsx to new product ID and pass down to the other components.

File: client/src/components/Related/ProductCard.jsx [hunk 2]
```diff
@@ -1,42 +1,66 @@
 import React, { useState, useEffect } from 'react';
```
Comment (cwg83) [inline client/src/components/Related/ProductCard.jsx:1]: This is the lowest-level component in my app - the related products state is mapped to this component, which displays the product image, name, price, and rating stars.

File: client/src/components/Related/Related.jsx [hunk 3]
```diff
@@ -8,9 +8,9 @@ const Related = (props) => {
   const [relatedProducts, setRelatedProducts] = useState();
```
Comment (cwg83) [inline client/src/components/Related/Related.jsx:8]: This is the top-most component of my module - it is basically a container for my RelatedProducts and YourOutfit components, and passes down necessary data as props.

File: client/src/components/Related/RelatedProducts.jsx [hunk 4]
```diff
@@ -1,20 +1,50 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
```
Comment (cwg83) [inline client/src/components/Related/RelatedProducts.jsx:1]: This is the container component for my related products - it maps the products from the relatedProducts state in Relatex.jsx and adds the data from each product to a ProductsCard.jsx component

File: client/src/components/Related/YourOutfit.jsx [hunk 5]
```diff
@@ -1,11 +1,43 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
```
Comment (cwg83) [inline client/src/components/Related/YourOutfit.jsx:1]: This is the component that displays items from the ourOutfit state in App.jsx (which I believe can be added from the Overview.jsx component). I'm still working on this one, so you probably won't see anything here in the browser yet.

File: package.json [hunk 6]
```diff
@@ -9,7 +9,7 @@
     "build:dev": "webpack --mode=development",
     "build:prod": "webpack --mode=production --node-env=production",
     "watch": "webpack --watch",
-    "serve": "webpack serve",
+    "serve": "webpack serve --mode=development",
```
Comment (cwg83) [inline package.json:12]: I forgot I added this, but this is what will set the MODE variable in our .env file to 'development' or 'production' so we can change our API call URLs based on which mode we are on.

File: client/src/components/Related/ProductCard.jsx (hunk 1 above)
Comment (sm-fm) [reply client/src/components/Related/ProductCard.jsx:14]: Nothing major to change here, but here are my thoughts --
- possibly move useEffect up to line 9 so it's clear how state is being updated upon mounting of this component
- get rid of the console.log on line 16
- move the "?" on line 46 to it's own line below like you did with the ":" on line 51 so it's easier to follow the ternary

Things you did well --
- your name conventions are really clear for variables and class names
- your functions are small and easy to follow

File: client/src/components/Related/Related.jsx [hunk 7]
```diff
@@ -8,9 +8,9 @@ const Related = (props) => {
...
   const [isFetching, setIsFetching] = useState(true);
 
-  const handleSetIsFetching = () => {
-    setIsFetching(!isFetching);
-  };
+  // const handleSetIsFetching = () => {
+  //   setIsFetching(!isFetching);
```
Comment (sm-fm) [inline client/src/components/Related/Related.jsx:12]: Reminder to remove comments if you don't need them anymore

File: client/src/components/Related/RelatedProducts.jsx [hunk 8]
```diff
@@ -1,20 +1,50 @@
...
+              yourOutfit={props.yourOutfit}
+              addToOutfit={props.addToOutfit}
+            />;
+          })
+          : null
+        }
 
-  )
-}
+      </div>
```
Comment (sm-fm) [inline client/src/components/Related/RelatedProducts.jsx:31]: This whole section is snazzy, very slick!

File: client/src/components/Related/RelatedProducts.jsx [hunk 9]
```diff
@@ -1,20 +1,50 @@
...
-      }
-    </div>
+          props.relatedProducts.map((product) => {
+            return <ProductCard
+              key={product.id}
+              product={product}
+              generateStars={props.generateStars}
+              isFetching={props.isFetching}
+              setIsFetching={props.setIsFetching}
+              parentComponent={componentName}
```
Comment (sm-fm) [inline client/src/components/Related/RelatedProducts.jsx:23]: I see you pass down the parent component name to ProductCard and you use that in ProductCard to conditionally render styles it looks like, but I don't understand exactly what's going on. Nothing to change, just curious to hear lol

**Review by sm-fm — COMMENTED** [review]
Overall looks really solid. Your naming conventions and small functions make it clear what's going on. I put comments throughout about possible things to change!

**Comment by cwg83** [comment]
@sm-fm all of your change suggestions made sense, I completed all of those changes and also fixed some linting issues.

**Pull request merged**